exasol_transformers_extension/udfs/model_downloader_udf.py

```diff
@@ -1,8 +1,40 @@
+import os
+import tempfile
+import transformers
+from pathlib import PurePosixPath
+from exasol_bucketfs_utils_python.bucketfs_factory import BucketFSFactory
 
 
 class ModelDownloader:
-    def __init__(self, exa):
+    def __init__(self, exa, downloader_method=transformers.AutoModel):
         self.exa = exa
+        self.downloader_method = downloader_method
 
     def run(self, ctx) -> None:
-        raise NotImplementedError
+        model_name = ctx.model_name
+        bfs_conn = ctx.bfs_conn
+
+        # set model path in buckets
+        model_path = model_name.replace('-', '_')
+
+        # create bucketfs location
+        bfs_conn_obj = self.exa.get_connection(bfs_conn)
+        bucketfs_location = BucketFSFactory().create_bucketfs_location(
+            url=bfs_conn_obj.address,
+            user=bfs_conn_obj.user,
+            pwd=bfs_conn_obj.password)
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            # download model into tmp folder
+            self.downloader_method.from_pretrained(
+                model_name, cache_dir=tmpdirname)
+
+            # upload the downloaded model files into bucketfs
+            for tmp_file_name in os.listdir(tmpdirname):
```

Comment:
do we only have files or also sub directories?

Reply:
maybe document the structure of the cache, also for the case of multiple models

Reply:
we have only files. 

These files are uploaded to bucketfs right after downloading to tmp directory. For this reason, I did not see the need to make an arrangement in the tmp directory. But there are separate directories for each model in bucketfs.